### PR TITLE
#162604730 View single Article functionality

### PR DESCRIPTION
**Pull request by sulenchy**
#### What does this PR do?
View single Article functionality

#### Description of Task to be completed?
- implement the functionality
- connect it to the landing page
- write test

#### How should this be manually tested?
- Clone the branch (feature/162604730/view-single-article-functionality)
- Run `npm install` to install the packages
- Run `npm run start:dev`

#### Any background context you want to provide?
N/A

#### What are the relevant pivotal tracker stories?

[#162604730](https://www.pivotaltracker.com/story/show/162604730)

#### Screenshots (if appropriate)

screen shot 2018-12-20 at 6 24 03 am
screen shot 2018-12-20 at 6 24 25 am

#### Questions:
N/A

#### Link to Review app:
[Review App](https://valinor-ah-frontend-stag-pr-29.herokuapp.com/)

### CODE REVIEW

File: src/actions/singleArticleActions.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+
+const fetchArticle = articleId => (dispatch) => {
+  dispatch(globalLoading(true));
+  return fetch(`${process.env.API_BASE_URL}/articles/${articleId}`)
+    .then(res => res.json())
+    .then((body) => {
+      if (body.errors) {
+        dispatch(globalFailure(body.errors));
+        toastr.error(body.status.toUpperCase(), body.errors.message[0]);
```
Comment (darthrighteous) [inline src/actions/singleArticleActions.js:20]: `errors.message` is not an array. It's a string, this will on print out the first character of the string

File: src/components/ArticlePage/index.js [hunk 2]
```diff
@@ -100,111 +106,129 @@ class ArticlePage extends React.Component {
...
+                    {article.body}
+                    <br />
+                    <br />
+                    {article.body}
+                    <br />
+                    <br />
+                    {article.body}
+                    <br />
+                    <br />
+                    {article.body}
```
Comment (darthrighteous) [inline src/components/ArticlePage/index.js]: I don't think we still need this repeating body, I only implemented it that way cause it was mockdata. It should show up only once.

File: src/components/common/header/NavBar.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React, { Component } from 'react';
 import more from '../../../../public/images/more.png';
 import mockCategories from '../../../../mockdata/categories';
 import NavItems from './NavItems';
-import UserSettings from './UserSettings';
+import { UserSettings } from './UserSettings';
```
Comment (darthrighteous) [inline src/components/common/header/NavBar.js]: what's the reason behind this change? isn't this now the unconnected component, is this intended?

File: src/components/common/header/UserSettings.js [hunk 4]
```diff
@@ -5,7 +5,7 @@ import { Link } from 'react-router-dom';
 
 import loginMobile from '../../../../public/images/login.png';
 import signup from '../../../../public/images/signup.png';
-import Login from './Login';
+import { Login } from './Login';
```
Comment (darthrighteous) [inline src/components/common/header/UserSettings.js]: what's the reason behind this change? isn't this now the unconnected component, is this intended?

File: tests/src/components/ArticlePage/CommentBox.test.js [hunk 5]
```diff
@@ -1,20 +1,41 @@
 import React from 'react';
 import { shallow } from 'enzyme';
-
 import CommentBox from '../../../../src/components/ArticlePage/CommentBox';
-import ArticlePage from '../../../../src/components/ArticlePage';
+import { ArticlePage } from '../../../../src/components/ArticlePage';
+import mockData from '../../../../mockdata/singleArticle';
+
+// const middlewares = [thunk];
+// const mockStore = configureMockStore(middlewares);
```
Comment (darthrighteous) [inline tests/src/components/ArticlePage/CommentBox.test.js]: you can go ahead to remove this if it's no longer needed

**Comment by darthrighteous** [comment]
@sulenchy the google sign up no longer works on the pr app, kindly look into this

File: tests/src/components/ArticlePage/CommentBox.test.js (hunk 5 above)
Comment (sulenchy) [reply tests/src/components/ArticlePage/CommentBox.test.js]: Thanks. Feedback duly observed. Consider it done

File: src/components/ArticlePage/index.js (hunk 2 above)
Comment (sulenchy) [reply src/components/ArticlePage/index.js]: feedback duly observed

File: src/components/common/header/UserSettings.js [hunk 6]
```diff
@@ -65,7 +65,11 @@ export class UserSettings extends Component {
 
 
 UserSettings.propTypes = {
-  isLoggedIn: PropTypes.bool.isRequired,
+  isLoggedIn: PropTypes.bool,
+};
+
+UserSettings.defaultProps = {
```
Comment (sulenchy) [inline src/components/common/header/UserSettings.js]: Reverse this change

File: src/components/ArticlePage/index.js (hunk 2 above)
Comment (sulenchy) [reply src/components/ArticlePage/index.js]: @Darthrighteous, the feedback has been implemented

File: src/components/common/header/NavBar.js (hunk 3 above)
Comment (sulenchy) [reply src/components/common/header/NavBar.js]: @Darthrighteous , Feedback duly observed and implemented

File: src/components/common/header/UserSettings.js (hunk 4 above)
Comment (sulenchy) [reply src/components/common/header/UserSettings.js]: @Darthrighteous , Feedback duly observed and  implemented
